### PR TITLE
Feat/contract info

### PR DESCRIPTION
**Pull request by FlorianHuc**
## PR description

Add info related to contracts created before London with nonce 0

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/frame/MessageFrame.java [hunk 1]
```diff
@@ -1169,6 +1172,27 @@ public Optional<ExceptionalHaltReason> getExceptionalHaltReason() {
     return exceptionalHaltReason;
   }
 
+  /**
+   * Gets extra data colelcted during the frame procesing.
```
Comment (macfarla) [inline evm/src/main/java/org/hyperledger/besu/evm/frame/MessageFrame.java:1176]:
```suggestion
   * Gets extra data collected during the frame processing.
```

File: evm/src/main/java/org/hyperledger/besu/evm/processor/ContractCreationProcessor.java [hunk 2]
```diff
@@ -87,6 +88,17 @@ private static boolean accountExists(final Account account) {
...
   }
 
+  /**
+   * check is it is empty as per London contract spec strictly.
+   *
+   * @param account to check
+   * @return true if it has a nonce == 0, empty code and non-empty storage. It means it's non-empty
+   *     and was created before London.
+   */
+  public static boolean contractHasNonceZeroEmptyCodeAndNonEmptyStorage(final Account account) {
```
Comment (macfarla) [inline evm/src/main/java/org/hyperledger/besu/evm/processor/ContractCreationProcessor.java:98]: needs a unit test

**Review by macfarla — COMMENTED** [review]
@FlorianHuc is there a spec or similar for this change? is this for linea to interop with geth?

**Comment by FlorianHuc** [comment]
Hi @macfarla we have discussed this PR with Fabio and I'll do differently. Closing it. Thank you.

**Pull request closed without merging**